src/goPackages.ts

```diff
@@ -2,54 +2,30 @@ import vscode = require('vscode');
 import cp = require('child_process');
 import path = require('path');
 import { getGoRuntimePath, getCurrentGoWorkspaceFromGOPATH } from './goPath';
-import { isVendorSupported, getCurrentGoPath, getToolsEnvVars } from './util';
-
-let allPkgs = new Map<string, string>();
-let goListAllCompleted: boolean = false;
-let goListAllPromise: Promise<Map<string, string>>;
-
-export function isGoListComplete(): boolean {
-	return goListAllCompleted;
-}
+import { isVendorSupported, getCurrentGoPath, getToolsEnvVars, getGoVersion, getBinPath, SemVersion } from './util';
 
 /**
  * Runs go list all
  * @returns Map<string, string> mapping between package import path and package name
  */
 export function goListAll(): Promise<Map<string, string>> {
-	let goRuntimePath = getGoRuntimePath();
-
-	if (!goRuntimePath) {
-		vscode.window.showInformationMessage('Cannot find "go" binary. Update PATH or GOROOT appropriately');
-		return Promise.resolve(null);
-	}
-
-	if (goListAllPromise) {
-		return goListAllPromise;
-	}
-
-	goListAllPromise = new Promise<Map<string, string>>((resolve, reject) => {
-		// Use `{env: {}}` to make the execution faster. Include GOPATH to account if custom work space exists.
-		const env: any = getToolsEnvVars();
-
-		const cmd = cp.spawn(goRuntimePath, ['list', '-f', '{{.Name}};{{.ImportPath}}', 'all'], { env: env, stdio: ['pipe', 'pipe', 'ignore'] });
+	return new Promise<Map<string, string>>((resolve, reject) => {
+		const cmd = cp.spawn(getBinPath('gopkgs'), ['-format', '{{.Name}};{{.ImportPath}}'], { env: getToolsEnvVars() });
 		const chunks = [];
 		cmd.stdout.on('data', (d) => {
 			chunks.push(d);
 		});
 
 		cmd.on('close', (status) => {
```

Comment:
Will check foe the missing tool when execution, but cannot detect failure on flag, because the old gopkgs doesn't care with any flags or arguments.

The only way to check it use different gopkgs version is by read the help provided by the binary.

Reply:
That's unfortunate